### PR TITLE
Add support for Observability for powerflex inject

### PR DESCRIPTION
**Pull request by sharmilarama**
# Description

This PR adds support to inject karavi-metrics-powerflex container with sidecar proxy.

# Issues

List the issues impacted by this PR:

| Issue ID |
| -------- |
|  |

# Checklist:

- [x] I have performed a self-review of my own changes.

### CODE REVIEW

**Comment by ioeicicdsvc** [comment]
@sharmilarama: Thank you for submitting your pull request!

Please make sure you have followed our contributing guidelines. We will review it as soon as possible.

------------------------------------------------------------------------------------------------------------------------
We want your feedback! If you have any questions or suggestions regarding our contributing process/workflow, please reach out to us at karavi@dell.com.

**Comment by atye** [comment]
Is it possible to add unit tests? Was injection tested with the latest powerflex driver?

File: cmd/karavictl/cmd/inject.go [hunk 1]
```diff
@@ -664,16 +664,21 @@ func (lc *ListChangeForPowerMax) injectIntoReverseProxy(imageAddr, proxyHost str
...
+	pluginID := deploy.Namespace
 
-	// Remove any existing proxy containers...
+	// Remove any existing proxy containers and check for observability containers...
 	for i, c := range containers {
 		if c.Name == "karavi-authorization-proxy" {
 			containers = append(containers[:i], containers[i+1:]...)
+		} else if c.Name == "karavi-metrics-powerflex" {
+			pluginID = "powerflex"
+		} else if c.Name == "karavi-metrics-powerstore" {
```
Comment (alikdell) [inline cmd/karavictl/cmd/inject.go:675]: Do we support powerstore in authorization?

Comment (sharmilarama) [reply cmd/karavictl/cmd/inject.go:675]: We have a story for that in backlog, it is targeted for 1.0

**Review by alikdell — APPROVED** [review]

**Review by LJ-Software — APPROVED** [review]

**Comment by sharmilarama** [comment]
> Is it possible to add unit tests? Was injection tested with the latest powerflex driver?

Unit Test data already covers powerflex metrics. Sidecar inject on 1.5.0 driver was tested successfully.

**Review by atye — APPROVED** [review]

**Pull request merged**